### PR TITLE
deployment pkg added With-hugepages,SC,Label funcs

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/deployment/deployment.go [hunk 1]
```diff
@@ -200,6 +200,89 @@ func (builder *Builder) WithSecondaryNetwork(networks []*multus.NetworkSelection
...
+// WithLabel applies label to pod's definition.
+func (builder *Builder) WithLabel(labelKey, labelValue string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(fmt.Sprintf("Defining deployment's label to %s:%s", labelKey, labelValue))
+
+	if labelKey == "" {
+		builder.errorMsg = "can not apply empty labelKey"
```
Comment (evgenLevin) [inline pkg/deployment/deployment.go:274]: add glog

**Comment by evgenLevin** [comment]
lgtm

File: pkg/deployment/deployment.go [hunk 2]
```diff
@@ -200,6 +200,91 @@ func (builder *Builder) WithSecondaryNetwork(networks []*multus.NetworkSelection
...
+		glog.V(100).Infof("The 'labelKey' of the deployment is empty")
+
+		builder.errorMsg = "can not apply empty labelKey"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.Template.Labels = map[string]string{labelKey: labelValue}
```
Comment (trewest) [inline pkg/deployment/deployment.go:283]: Should you overwrite rather than just append?

File: pkg/deployment/deployment.go [hunk 3]
```diff
@@ -200,6 +200,91 @@ func (builder *Builder) WithSecondaryNetwork(networks []*multus.NetworkSelection
...
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.Template.Spec.SecurityContext = securityContext
+
+	return builder
+}
+
+// WithLabel applies label to pod's definition.
```
Comment (trewest) [inline pkg/deployment/deployment.go:265]: nit: deployment

File: pkg/deployment/deployment.go (hunk 2 above)
Comment (kononovn) [reply pkg/deployment/deployment.go]: Change function a bit. Please review 1 more time

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**